### PR TITLE
[MIG] hr_department_sequence: Migration to 10.0

### PR DESCRIPTION
**Pull request by eric-lemesre**

### CODE REVIEW

**Comment by feketemihai** [comment]
@eric-lemesre please follow migration instructions from #259 and please check Odoo code base, in essence you can remove parents and childs fields since they added in hr module, leaving just code and sequence.

File: hr_department_sequence/models/hr_department.py [hunk 1]
```diff
@@ -32,13 +32,8 @@ class HrDepartment(models.Model):
     sequence = fields.Integer(string='Sequence', index=True,
                               help="Gives the sequence order when displaying "
                               "a list of departments.")
-    parent_id = fields.Many2one(ondelete='restrict')
-    parent_left = fields.Integer(string='Left Parent', index=True)
-    parent_right = fields.Integer(string='Right Parent', index=True)
-    child_ids = fields.One2many(comodel_name='hr.department',
-                                inverse_name='parent_id',
-                                string='Children Departments')
-    active = fields.Boolean(string='Active', default=True)
```
Comment (dreispt) [inline hr_department_sequence/models/hr_department.py:41]: Why remove these fields?

File: hr_department_sequence/views/hr_department_view.xml [hunk 2]
```diff
@@ -35,7 +35,7 @@
             <field name="model">hr.department</field>
             <field name="inherit_id" ref="hr.view_department_tree"/>
             <field name="arch" type="xml">
-                <xpath expr="//field[@name='complete_name']" position="after">
+                <xpath expr="//field[@name='display_name']" position="after">
```
Comment (dreispt) [inline hr_department_sequence/views/hr_department_view.xml:38]: Why this change?

File: hr_department_sequence/models/hr_department.py (hunk 1 above)
Comment (eric-lemesre) [reply hr_department_sequence/models/hr_department.py:41]: Because this fields are already in Odoo core  https://github.com/OCA/OCB/blob/5654fe4dd3c4ae55fe13bac89cd1790b6d1593af/addons/hr/models/hr.py#L280

File: hr_department_sequence/views/hr_department_view.xml (hunk 2 above)
Comment (eric-lemesre) [reply hr_department_sequence/views/hr_department_view.xml:38]: This change needs more work in hr_department V10.0 the name his name, and in V11 this is complete_name

Comment (eric-lemesre) [reply hr_department_sequence/views/hr_department_view.xml:38]: This is field of displayed name of company. 
https://github.com/OCA/OCB/blob/9d371705228bf58ea9a17255d7677666cb2a3479/addons/hr/views/hr_views.xml#L480

File: hr_department_sequence/models/hr_department.py (hunk 1 above)
Comment (dreispt) [reply hr_department_sequence/models/hr_department.py:41]: ok

**Comment by dreispt** [comment]
Looks good, but build is failing: "The field parent_id on model hr.department must be set as ondelete='cascade' or 'restrict'"
